proxy/server.py

```diff
@@ -44,768 +44,1023 @@
+                    localStorage.setItem("config-details-open", details.open);
+                });
+            });
+        </script>
+        """
+        message += "</table>\n"
+        message += f'\n<p>Page refresh: {str(datetime.datetime.fromtimestamp(time.time()))}</p>\n</body>\n</html>'
+        return self.wfile.write(message.encode(UTF_8))
+
+
+    def handle_problems(self) -> str:
+        return self.send_json_response({"problems": []})
+
+    def handle_allowlist(self, path) -> str:
+        response = self.pw.poll(path, jsonformat=True)
+        return self.send_json_response(json.loads(response))
+
+    def handle_tedapi(self, path) -> str:
+        if not self.pw.tedapi:
+            return self.send_json_response({"error": "TEDAPI not enabled"}, status_code=HTTPStatus.BAD_REQUEST)
+
+        commands = {
+            '/tedapi/config': self.pw.tedapi.get_config,
+            '/tedapi/status': self.pw.tedapi.get_status,
+            '/tedapi/components': self.pw.tedapi.get_components,
+            '/tedapi/battery': self.pw.tedapi.get_battery_blocks,
+            '/tedapi/controller': self.pw.tedapi.get_device_controller,
+        }
+        command = commands.get(path)
+        if command:
+            return self.send_json_response(command())
+
+        return self.send_json_response(
+            {"error": "Use /tedapi/config, /tedapi/status, /tedapi/components, /tedapi/battery, /tedapi/controller"},
+            status_code=HTTPStatus.BAD_REQUEST
+        )
+
+
+    def handle_cloud(self, path) -> str:
+        if not self.pw.cloudmode or self.pw.fleetapi:
+            return self.send_json_response({"error": "Cloud API not enabled"}, status_code=HTTPStatus.BAD_REQUEST)
+
+        commands = {
+            '/cloud/battery': self.pw.client.get_battery,
+            '/cloud/power': self.pw.client.get_site_power,
+            '/cloud/config': self.pw.client.get_site_config,
+        }
+        command = commands.get(path)
+        if command:
+            return self.send_json_response(command())
+        return self.send_json_response({"error": "Use /cloud/battery, /cloud/power, /cloud/config"}, status_code=HTTPStatus.BAD_REQUEST)
+
+
+    def handle_fleetapi(self, path) -> str:
+        if not self.pw.fleetapi:
+            return self.send_json_response({"error": "FleetAPI not enabled"}, status_code=HTTPStatus.BAD_REQUEST)
+
+        commands = {
+            '/fleetapi/info': self.pw.client.get_site_info,
+            '/fleetapi/status': self.pw.client.get_live_status,
+        }
+        command = commands.get(path)
+        if command:
+            return self.send_json_response(command())
+
+        return self.send_json_response({"error": "Use /fleetapi/info, /fleetapi/status"}, status_code=HTTPStatus.BAD_REQUEST)
+
+
+    def handle_static_content(self, path) -> str:
+        self.proxystats[PROXY_STATS_TYPE.GETS] += 1
+        self.send_response(HTTPStatus.OK)
+        self.send_header('Content-type', 'text/html')
```

Comment:
You can't sent this here. The static content could be of a differen type. Just remove this and allow the content type to be set latter on after the static content is determined.

Reply:
Removed.